qiskit/visualization/utils.py

```diff
@@ -272,6 +280,25 @@ def get_condition_label(condition, clbits, bit_locations, cregbundle):
     return label, clbit_mask, vlist
 
 
+def fix_special_characters(label, size):
+    """Convert any special characters for mpl and latex drawers
+
+    Args:
+        label (str): the label to fix
+        size (int): if a register, the size
+        cregbundle (bool): if set True classical registers are bundled
+
+    Returns:
+        str: label to display
+    """
+    # Currently only checks for multiple underscores in register names
+    rep_count = label.count("_")
+    if size > 1:    # for subscripts at end, don't replace "_"
+        rep_count -= 1
+    label = label.replace("_", r"\_", rep_count).replace(" ", "\\;")
+    return label
```

Comment:
If it's reasonable within the drawers, I'd be tempted to have the "escape" function applied only to the registers, and then let the drawers tack on a suffix after this function.  Unless I'm missing something, it seems odd to add on some characters that are _meant_ to be special, and then have this function need to re-parse them specifically to _not_ escape them.

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/7459/commits/1fa466eb5b68a6c00ed2dd5e7c1669b097ac6a0d.